patch/0001-fix-os-crash-caused-by-optoe-when-class-switch.patch

```diff
@@ -0,0 +1,36 @@
+From b8ba60c6d604fee3f61e3d1e709e86c785b075f4 Mon Sep 17 00:00:00 2001
+From: philo <philo@micasnetworks.com>
+Date: Mon, 1 Jul 2024 11:03:35 +0800
+Subject: [PATCH] fix os crash caused by optoe when class switch
+
+---
+ drivers/misc/eeprom/optoe.c | 5 ++++-
+ 1 file changed, 4 insertions(+), 1 deletion(-)
+
+diff --git a/drivers/misc/eeprom/optoe.c b/drivers/misc/eeprom/optoe.c
+index 9129312a1..a6bdbb9d1 100644
+--- a/drivers/misc/eeprom/optoe.c
++++ b/drivers/misc/eeprom/optoe.c
+@@ -941,8 +941,10 @@ static ssize_t set_dev_class(struct device *dev,
+ 	} else {
+ 		/* one-address (eg QSFP) and CMIS family */
+ 		/* if it exists, remove 0x51 i2c address */
+-		if (optoe->client[1])
++		if (optoe->client[1]) {
+ 			i2c_unregister_device(optoe->client[1]);
++			optoe->client[1] = NULL;
++		}
+ 		optoe->bin.size = ONE_ADDR_EEPROM_SIZE;
+ 		optoe->num_addresses = 1;
+ 	}
+@@ -1226,6 +1228,7 @@ static int optoe_probe(struct i2c_client *client,
+ 	if (num_addresses == 2) {
+ 		if (optoe->client[1])
+ 			i2c_unregister_device(optoe->client[1]);
++			optoe->client[1] = NULL;
```

Comment:
Missing braces around the if-block.

Reply:
@saiarcot895  Has updated. Please help to approve and merge, thanks!